lib/logscheduler.c

```diff
@@ -229,21 +240,25 @@ _queue_thread(LogScheduler *self, LogSchedulerThreadState *thread_state, LogMess
   if (thread_state->num_messages == 0)
     main_loop_worker_register_batch_callback(&thread_state->batch_callback);
 
-  guint partition_index = _get_partition_index(self, thread_state, msg);
+  gboolean should_flush;
+  guint partition_index = _get_partition_index(self, thread_state, msg, &should_flush);
 
   LogMessageQueueNode *node;
   node = log_msg_alloc_queue_node(msg, path_options);
   iv_list_add_tail(&node->list, &thread_state->batch_by_partition[partition_index]);
   thread_state->num_messages++;
   log_msg_unref(msg);
+
+  if (should_flush)
+    _move_batch_to_partition(self, thread_state, partition_index);
```

Comment:
This actually degrades performance. I can't explain why (maybe lock contention?), but if we assume that  input threads are an order of magnitude faster than the processing threads, this extra flush shouldn't matter.

Reply:
Reverted